sdk/messaging/azservicebus/receiver.go

```diff
@@ -361,6 +369,12 @@ func (r *Receiver) DeadLetterMessage(ctx context.Context, message *ReceivedMessa
 }
 
 func (r *Receiver) receiveMessagesImpl(ctx context.Context, maxMessages int, options *ReceiveMessagesOptions) ([]*ReceivedMessage, error) {
+	var err error
```

Comment:
If you move this span code into `ReceiveMessages` it'll be less tricky since you'll only have a single spot where you need to log the error, and we don't have to be wary of assigning `err` in all the code here.

Reply:
updated!